src/database-controller/src/poller/index.js

```diff
@@ -95,6 +100,35 @@ function synchronizeHandler(snapshot, addOns, pollingTs) {
         );
       }),
     )
+    .catch(err => {
+      // if we are sure the error is not recoverable, we will mock a failed framework, and record this in events.
+      if (_.has(err, 'response') && isUnrecoverableResponse(err.response)) {
+        logger.warn(
+          `An error happened when synchronize request for framework ${frameworkName} and pollingTs=${pollingTs}. We are sure the error is not recoverable. Will mock a failed framework.`,
+          err,
+        );
+        queue.add(async () => {
+          // For safety reason, we only consider the job that never starts.
+          // snapshot.setFailed() will raise an error if the framework is not in state AttemptCreationPending or has retry history.
+          snapshot.setFailed();
+          await postMockedEvent(snapshot, 'MODIFIED');
+          await databaseModel.FrameworkEvent.create(
+            generateClusterEventUpdate(
+              snapshot,
+              'Warning',
+              'UnrecoverableSynchronizeFailure',
```

Comment:
>UnrecoverableSynchronizeFailure [](start = 15, length = 31)

Hard to understand, align to exit spec phase

Reply:
done